astropy/time/formats.py

```diff
@@ -1242,6 +1306,29 @@ def parse_string(self, timestr, subfmts):
 
     def set_jds(self, val1, val2):
         """Parse the time strings contained in val1 and set jd1, jd2"""
+        # If specific input subformat is required then use the Python parser.
+        # Also do this if Time format class does not define `use_fast_parser` or
+        # if the fast parser is entirely disabled. Note that `use_fast_parser`
+        # is ignored for format classes that don't have a fast parser.
+        if (self.in_subfmt != '*'
```

Comment:
Could one reverse the logic?
```
if (self.in_subfmt == '*'
        and 'fast_parser_pars' in self.__class__.__dict__
        and conf.use_fast_parser != 'False'):
    try:
        self.jd1, self.jd2 = self.get_jds_fast(val1, val2)
        return
    except Exception:
        # Fall through to the Python parser unless fast is forced.
        if conf.use_fast_parser == 'force':
            raise

self.jd1, self.jd2 = self.get_jds_python()
return
```

Reply:
I agree this *looks* cleaner.  But I dropped this in and it failed most tests. Since the original is working I'll leave it.